examples/interpreter-demo/serialize.js

```diff
@@ -0,0 +1,411 @@
+/**
+ * @license
+ * Copyright 2017 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Saving and restoring the state of a JS-Intepreter.
+ * @author fraser@google.com (Neil Fraser)
+ */
+'use strict';
+
+
+// Constructors for objects within Acorn.
+var NODE_CONSTRUCTOR;
+var NODE_LOC_CONSTRUCTOR;
+var LINE_LOC_CONSTRUCTOR;
+
+function recordAcornConstructons(interpreter) {
+  // Constructors for objects within Acorn.
+  NODE_CONSTRUCTOR = interpreter.ast.constructor;
+  NODE_LOC_CONSTRUCTOR = interpreter.ast.loc &&
+      interpreter.ast.loc.constructor;
+  LINE_LOC_CONSTRUCTOR = interpreter.ast.loc &&
+      interpreter.ast.loc.end.constructor;
+}
+
+function deserialize(json, interpreter) {
+  function decodeValue(value) {
```

Comment:
This function can be defined at the top level if you pass in `objectList`.

Reply:
While this function can be trivially broken out, I think it's best for it to exactly mirror its `encodeValue` counterpart which cannot be easily broken out (pending comment below).

Reply:
Done.